Improve influxdb roles; Use Ansible 2.8 syntax shortcut; A few other fixes

ypid opened the pull request:
Closes: #1371

I reviewed and tested this with all the use cases I have. But I don’t use all roles so I have not tested everything. The regex I came up with does quite a bit of validation so it does not blindly replace stuff. The one known issue is that opening/closing parenthesis `(`, `)` are maybe not removed together, leaving one there which is a syntax error. Please test this a bit with your use cases before you merge. Other than this, I hope I fixed more bugs than I introduce with this.

ypid committed:
[influxdb_server] Apply ShellCheck suggestions and DebOps standards

ypid committed:
[influxdb] Remove useless newline

ypid committed:
[influxdb_server] Use Ansible 2.8 default shortcut

ypid committed:
[docs] Fix spelling

ypid committed:
[debops-tools] Improve debops-optimize-old a bit

ypid committed:
[debops-tools] debops-optimize (ng) Use Ansible 2.8 default shortcut

ypid committed:
First attempt at a tool assistant use Ansible 2.8 shortcut update

Ref: https://github.com/debops/debops/issues/1371

Mechanical edit done by:

```Shell
./bin/debops-optimize
```

ypid committed:
[influxdb_server] autoinfluxdbbackup is not nounset ready

Partly reverts: dbb0069d97d1bee487799fbc1aa1346a94d75ca6

ypid committed:
[docs] Clarify that version number refers to Ansible version

ypid committed:
[influxdb] Comply with DebOps standards

ypid committed:
[influxdb_server] Make generated influxdb.conf look prettier

No need for two empty lines at the top.

ypid committed:
I found the holy grail of tool assistant Ansible 2.8 shortcut update

ypid committed:
Tool assistant use Ansible 2.8 shortcut update

Mechanical edit done only by:

```Shell
./bin/debops-optimize
```

ypid committed:
Manual post work after tool assisted edit (Ansible 2.8 shortcut)

ypid committed:
Optimize the debops-optimize based on experience of manual post work

ypid committed:
Rerun debops-optimize to let it update defaults of defaults

ypid committed:
[debops-tools] Optimize debops-optimize to find more Ansible 2.8 shortc.

ypid committed:
Tool assistant use Ansible 2.8 shortcut update

Mechanical edit done only by:

```Shell
./bin/debops-optimize
```

ypid committed:
Manually switch to Ansible 2.8 shortcut in more files

Looking for more easy to recognize for a machine and common cases.
Not found any so far.

ypid committed:
[debops-tools] Optimize debops-optimize to find more Ansible 2.8 shortc.

ypid committed:
Tool assistant use Ansible 2.8 shortcut update

ypid committed:
Manual post work after tool assisted edit (Ansible 2.8 shortcut)

ypid committed:
[mariadb, mariadb_server] Fix self ref ansible_local.mariadb.flavor

Uncovered by validating regular expression in debops-optimize.

ypid committed:
[debops-tools] Optimize debops-optimize to find more Ansible 2.8 shortc.

ypid committed:
Fix issues introduced by debops-optimize (Ansible 2.8 shortcut)

ypid committed:
Fix issues introduced by debops-optimize (Ansible 2.8 shortcut)

Mechanical edit done by:

```Shell
git ls-files -z "$(git rev-parse --show-toplevel)" | xargs --null -I '{}' find './{}' -type f -print0 | xargs --null sed --in-place --regexp-extended 's~d(ansible_local.core.unsafe_writes\|d\(\))~d(\1)~g;'
```

ypid committed:
Fix issues introduced by debops-optimize (Ansible 2.8 shortcut)

ypid committed:
[dnsmasq] No DNS INPUT firewall rule created for "listen to all"

ypid committed:
Manually switch to Ansible 2.8 shortcut in more files

ypid committed:
[debops-tools] Optimize debops-optimize to find more Ansible 2.8 shortc.

ypid committed:
[mosquitto] Use Ansible 2.8 shortcut

ypid committed:
Manually switch to Ansible 2.8 shortcut in more files

ypid committed:
[mosquitto] python3-paho-mqtt became available with Debian Buster

ypid committed:
[ifupdown] Wrap lines longer than 79 characters

ypid committed:
[mosquitto] Use debops.python role to handle Python dependencies

drybjed commented on ansible/roles/apt_cacher_ng/defaults/main.yml:
```diff
@@ -201,12 +201,8 @@ apt_cacher_ng__max_download_speed_kib: ''
-                                               ansible_local.apt.default_sources_map.Debian|d() and
-                                               ansible_local.apt.default_sources_map.Debian[0]|d())
-                                           else "http://deb.debian.org/debian" }}'
+apt_cacher_ng__upstream_mirror_debian: '{{ ansible_local.apt.default_sources_map.Dsdfebian[0]
```
Spelling mistake (`Dsdfebian`).

ypid replied:
That was me testing if `d()` also works with array access and then forgetting to change it back before committing (and failing to self review my own commit changes). Thanks for your careful review!

drybjed commented on ansible/roles/dnsmasq/tasks/main.yml:
```diff
@@ -29,8 +29,7 @@
     mode: '0755'
-    unsafe_writes: '{{ True if (core__unsafe_writes|d(True if (ansible_local|d() and ansible_local.core|d()
-                       and ansible_local.core.unsafe_writes|d() | bool) else False) | bool) else omit }}'
+    unsafe_writes: '{{ True if (core__unsafe_writes|d(ansible_local.core.unsafe_writes|d()) | bool) else omit }}'
```
Well... I'm not sure why I didn't notice this usage before (the default variable is referenced first instead of the fact) but this is essentially redundant.

Actually the use of the `ansible_local.core.*` facts in various roles could probably be replaced to references to variables defined in the inventory directly. The information flow would also be better, instead of getting the value from the remote host via facts, it could be taken from the inventory. The reason I didn't do this is that there wasn't a good variable namespace to reserve for this. But I think that after renaming the `debops` role to `controller`, the `debops__` namespace could be reserved for global variables, like `debops__unsafe_writes` and similar. Then we could drop most of the stuff in the `core` role facts and rely on the users to define the variables in the inventory. Then the above code could be changed to:
```yaml
unsafe_writes: '{{ True if debops__unsafe_writes|bool else omit }}'
```
As for the current cases in the file - I guess you can leave them as they are for now.

ypid replied:
Sounds good. `debops__` works for me. Then I will leave this as is for now.

ypid committed:
[apt_cacher_ng] Fix broken variable

This was a manual test case to see how the default variable would behave
with array access. Sadly, I committed this.

Thanks to @drybjed for the very careful review.

drybjed reviewed (approved):
All files reviewed. Nice job, thanks!

ypid commented:
Thanks for reviewing this so quickly :+1: 

Also thanks for fixing the remaining issues. For reference, https://github.com/debops/debops/pull/1386, https://github.com/debops/debops/pull/1385 and https://github.com/debops/debops/pull/1384.

drybjed commented:
I wish that I had more scrutiny before checking this PR, it added hours of testing and fixing bugs, and since I'm currently using my workstation for the CI pipeline, I cannot use it during the full test due to the load which takes hours. Oh well, new release will probably be over the weekend after I fix all the bugs...

ypid commented:
I warned you ;-) :

> Please test this a bit with your use cases before you merge.

And sorry for the work it caused. At least we now have much more clean `ansible_local` usage.
